src/index.ts

```diff
@@ -16,6 +16,15 @@ const fromNullable = <T>(v: T | Nullable): Maybe<T> => {
   return nothing();
 };
 
+/**
+ * Converts a value that _may_ be empty, such as a string or an array, to a
+ * maybe. It will accept anything that responds to length.
+ * If the value is has a length of 0, then Nothing is returned.
+ * If the value is anything else, it is returned wrapped in a Just.
+ */
+const fromEmpty = <T extends { length: number }>(xs: T): Maybe<T> =>
```

Comment:
Let's alias this `{ length: number }` so we can attach a name to this concept.
```
type Emptyable = { length: number };
```
or make it an interface
```
interface Emptyable {
  length: number;
}
```
Put that near the top of the file, after Nullable.